### DISCUSSION TITLE
Bug - TypeError: compRef.setInput is not a function

### DISCUSSION THREAD

**Discussion by catherinekennedy** [gund/ng-dynamic-component#499] [Q&A] [answered]
![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/67736813/218058639-6a06460d-01d2-47ce-9a90-20ba79a2182c.png)

<ndc-dynamic  [ngComponentOutlet]="item['componentType']" [ndcDynamicInputs]="item['inputs']" ></ndc-dynamic >

whenever I write this code the  ndcDynamicInputs is not working and shows the errorr like this:
TypeError: compRef.setInput is not a function

how to solve this error.

**Comment by gund** [comment, marked as answer]
`compRef.setInput` is a new Angular v14 API so it seems that you are running latest version of this library but not latest Angular version?

You can refer to the [compatibility with Angular table](https://github.com/gund/ng-dynamic-component#ng-dynamic-component) for more info.
